### ISSUE TITLE
eval and predict get the different result with same dataset and same checkpoint

### ISSUE THREAD

**Issue by bigzt** [tensorflow/tensor2tensor#1552]
### Description

hi there, I'm training a network to predict a time-series classification. Each target timestamp is a classification and irrelevant to the previous one. I use a py_func to print the predictions in a customed metrics function, but the results are different between t2t_decode (greedy d) with the same test dataset which created by t2t_datagen. I checked the inputs, is the same (if eval keeps the input sequence). The eval results are much accuracy than the predict ones.
what did I wrong?

PREDICT LOG

```
INFO:tensorflow:Importing user module src from path ./src
WARNING:tensorflow:From ./tf_env/lib/python3.6/site-packages/tensor2tensor/utils/trainer_lib.py:240: RunConfig.__init__ (from tensorflow.contrib.learn.python.learn.estimators.run_config) is deprecated and will be removed in a future version.
Instructions for updating:
When switching to tf.estimator.Estimator, use tf.estimator.RunConfig instead.
INFO:tensorflow:Configuring DataParallelism to replicate the model.
INFO:tensorflow:schedule=continuous_train_and_eval
INFO:tensorflow:worker_gpu=1
INFO:tensorflow:sync=False
WARNING:tensorflow:Schedule=continuous_train_and_eval. Assuming that training is running on a single machine.
INFO:tensorflow:datashard_devices: ['gpu:0']
INFO:tensorflow:caching_devices: None
INFO:tensorflow:ps_devices: ['gpu:0']
INFO:tensorflow:Using config: {'_task_type': None, '_task_id': 0, '_cluster_spec': <tensorflow.python.training.server_lib.ClusterSpec object at 0x7fe633d77f98>, '_master': '', '_num_ps_replicas': 0, '_num_worker_replicas': 0, '_environment': 'local', '_is_chief': True, '_evaluation_master': '', '_train_distribute': None, '_eval_distribute': None, '_device_fn': None, '_tf_config': gpu_options {
  per_process_gpu_memory_fraction: 1.0
}
, '_tf_random_seed': None, '_save_summary_steps': 100, '_save_checkpoints_secs': None, '_log_step_count_steps': 100, '_protocol': None, '_session_config': gpu_options {
  per_process_gpu_memory_fraction: 0.95
}
allow_soft_placement: true
graph_options {
  optimizer_options {
    global_jit_level: OFF
  }
}
isolate_session_state: true
, '_save_checkpoints_steps': 1000, '_keep_checkpoint_max': 20, '_keep_checkpoint_every_n_hours': 10000, '_model_dir': './src/t2t_model/eval/model.ckpt-313000', 'use_tpu': False, 't2t_device_info': {'num_async_replicas': 1}, 'data_parallelism': <tensor2tensor.utils.expert_utils.Parallelism object at 0x7fe633d77fd0>}
WARNING:tensorflow:Estimator's model_fn (<function T2TModel.make_estimator_model_fn.<locals>.wrapping_model_fn at 0x7fe63cf722f0>) includes params argument, but params are not passed to Estimator.
INFO:tensorflow:decode_hp.batch_size not specified; default=256
WARNING:tensorflow:From ./tf_env/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
2019_03_28:   4%|#1                         | 197/4646 [00:01<00:34, 130.09it/s]WARNING:tensorflow:From ./tf_env/lib/python3.6/site-packages/tensor2tensor/utils/decoding.py:616: py_func (from tensorflow.python.ops.script_ops) is deprecated and will be removed in a future version.
Instructions for updating:
tf.py_func is deprecated in TF V2. Instead, use
    tf.py_function, which takes a python function which manipulates tf eager
    tensors instead of numpy arrays. It's easy to convert a tf eager tensor to
    an ndarray (just call tensor.numpy()) but having access to eager tensors
    means tf.py_function s can use accelerators such as GPUs as well as
    being differentiable using a gradient tape.
   
WARNING:tensorflow:Input graph does not use tf.data.Dataset or contain a QueueRunner. That means predict yields forever. This is probably a mistake.
INFO:tensorflow:Calling model_fn.
INFO:tensorflow:Unsetting shared_embedding_and_softmax_weights.
INFO:tensorflow:Setting T2TModel mode to 'infer'
INFO:tensorflow:Setting hparams.dropout to 0.0
INFO:tensorflow:Setting hparams.label_smoothing to 0.0
INFO:tensorflow:Setting hparams.layer_prepostprocess_dropout to 0.0
INFO:tensorflow:Setting hparams.symbol_dropout to 0.0
INFO:tensorflow:Setting hparams.attention_dropout to 0.0
INFO:tensorflow:Setting hparams.relu_dropout to 0.0
INFO:tensorflow:Setting hparams.dropconnect_dropout to 0.0
INFO:tensorflow:Greedy Decoding
INFO:tensorflow:Using variable initializer: uniform_unit_scaling
WARNING:tensorflow:From ./tf_env/lib/python3.6/site-packages/tensor2tensor/utils/t2t_model.py:1358: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
INFO:tensorflow:Transforming feature 'inputs' with real_l2_loss_modality.bottom
WARNING:tensorflow:From ./tf_env/lib/python3.6/site-packages/tensor2tensor/layers/modalities.py:370: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING:tensorflow:From ./tf_env/lib/python3.6/site-packages/tensor2tensor/layers/modalities.py:370: dense (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.dense instead.
INFO:tensorflow:Transforming feature 'targets' with symbol_modality_18_128.targets_bottom
WARNING:tensorflow:From ./tf_env/lib/python3.6/site-packages/tensorflow/python/framework/function.py:1007: calling Graph.create_op (from tensorflow.python.framework.ops) with compute_shapes is deprecated and will be removed in a future version.
Instructions for updating:
Shapes are always computed; don't use the compute_shapes as it has no effect.
INFO:tensorflow:Building model body
WARNING:tensorflow:From ./tf_env/lib/python3.6/site-packages/tensor2tensor/models/research/universal_transformer.py:78: calling dropout (from tensorflow.python.ops.nn_ops) with keep_prob is deprecated and will be removed in a future version.
Instructions for updating:
Please use `rate` instead of `keep_prob`. Rate should be set to `rate = 1 - keep_prob`.
INFO:tensorflow:Transforming body output with symbol_modality_18_128.top
WARNING:tensorflow:From ./tf_env/lib/python3.6/site-packages/tensor2tensor/utils/t2t_model.py:1205: to_int64 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Graph was finalized.
2019-04-21 05:54:04.543271: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-04-21 05:54:04.641383: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:998] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-04-21 05:54:04.642082: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x36b1840 executing computations on platform CUDA. Devices:
2019-04-21 05:54:04.642097: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): GeForce GTX 1080 Ti, Compute Capability 6.1
2019-04-21 05:54:04.663444: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3393175000 Hz
2019-04-21 05:54:04.664394: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x56cfae0 executing computations on platform Host. Devices:
2019-04-21 05:54:04.664434: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
2019-04-21 05:54:04.664992: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1433] Found device 0 with properties: 
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.607
pciBusID: 0000:09:00.0
totalMemory: 10.91GiB freeMemory: 10.53GiB
2019-04-21 05:54:04.665024: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0
2019-04-21 05:54:04.666238: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-04-21 05:54:04.666257: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0 
2019-04-21 05:54:04.666266: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N 
2019-04-21 05:54:04.666728: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10616 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:09:00.0, compute capability: 6.1)
WARNING:tensorflow:From ./tf_env/lib/python3.6/site-packages/tensorflow/python/training/saver.py:1266: checkpoint_exists (from tensorflow.python.training.checkpoint_management) is deprecated and will be removed in a future version.
Instructions for updating:
Use standard file APIs to check for files with this prefix.
INFO:tensorflow:Restoring parameters from ./t2t_model/eval/model.ckpt-313000
INFO:tensorflow:Running local_init_op.
```

### Environment information

OS: ubuntu 18.04
tensorflow 1.13.1
tensor2tensor 1.13.2

$ pip freeze | grep tensor
# your output here

$ python -V
# your output here
```

### For bugs: reproduction and error logs

```
# Steps to reproduce:
...
```

```
# Error logs:
...
```
